test/metabase/test/data.clj

```diff
@@ -285,3 +287,50 @@
   [dataset & body]
   `(with-temp-db [_# (resolve-dbdef '~dataset)]
      ~@body))
+
+(defn delete-model-instance!
+  "Allows deleting a row by the model instance toucan returns when
+  it's inserted"
+  [{:keys [id] :as instance}]
+  (db/delete! (-> instance name symbol) :id id))
+
+(defn call-with-data
```

Comment:
This one is only used by the `with-data` macro, but when the macro expands, clojure complains about `call-with-data` being private, so I'll leave this as public.